### ISSUE TITLE
uint16, uint32 comparisons throw errors

### ISSUE THREAD

**Issue by zabir-nabil** [tensorflow/tensorflow#39457]
<em>Some datatypes such as uint16, uint32 can't be compared.</em>

**System information**
Ran on google colab.

**Describe the current behavior**
tensors with dtypes uint16, uint32 can't be compared with any other datatypes.

**Describe the expected behavior**
comparison for these datatypes should be supported/

**Standalone code to reproduce the issue**
```
import tensorflow as tf

ones = tf.ones((2, 3), dtype = tf.uint32)
zeros = tf.zeros((2, 3), dtype = tf.uint32)
tf.math.equal(ones, zeros)
```
Error log:
```
---------------------------------------------------------------------------

NotFoundError                             Traceback (most recent call last)

<ipython-input-62-2f39e3ae3672> in <module>()
      3 ones = tf.ones((2, 3), dtype = tf.uint32)
      4 zeros = tf.zeros((2, 3), dtype = tf.uint32)
----> 5 tf.math.equal(ones, zeros)

4 frames

/usr/local/lib/python3.6/dist-packages/six.py in raise_from(value, from_value)

NotFoundError: Could not find valid device for node.
Node:{{node Equal}}
All kernels registered for op Equal :
  device='XLA_GPU'; T in [DT_FLOAT, DT_DOUBLE, DT_INT32, DT_UINT8, DT_INT16, ..., DT_QUINT8, DT_QINT32, DT_BFLOAT16, DT_COMPLEX128, DT_HALF]
  device='XLA_CPU_JIT'; T in [DT_FLOAT, DT_DOUBLE, DT_INT32, DT_UINT8, DT_INT16, ..., DT_QUINT8, DT_QINT32, DT_BFLOAT16, DT_COMPLEX128, DT_HALF]
  device='XLA_GPU_JIT'; T in [DT_FLOAT, DT_DOUBLE, DT_INT32, DT_UINT8, DT_INT16, ..., DT_QUINT8, DT_QINT32, DT_BFLOAT16, DT_COMPLEX128, DT_HALF]
  device='GPU'; T in [DT_BOOL]
  device='GPU'; T in [DT_COMPLEX128]
  device='GPU'; T in [DT_COMPLEX64]
  device='GPU'; T in [DT_INT64]
  device='GPU'; T in [DT_INT16]
  device='GPU'; T in [DT_INT8]
  device='GPU'; T in [DT_INT32]
  device='GPU'; T in [DT_UINT8]
  device='GPU'; T in [DT_DOUBLE]
  device='GPU'; T in [DT_HALF]
  device='GPU'; T in [DT_FLOAT]
  device='CPU'; T in [DT_BOOL]
  device='CPU'; T in [DT_STRING]
  device='CPU'; T in [DT_COMPLEX128]
  device='CPU'; T in [DT_COMPLEX64]
  device='CPU'; T in [DT_INT64]
  device='CPU'; T in [DT_INT32]
  device='CPU'; T in [DT_BFLOAT16]
  device='CPU'; T in [DT_INT16]
  device='CPU'; T in [DT_INT8]
  device='CPU'; T in [DT_UINT8]
  device='CPU'; T in [DT_DOUBLE]
  device='CPU'; T in [DT_HALF]
  device='CPU'; T in [DT_FLOAT]
  device='XLA_CPU'; T in [DT_FLOAT, DT_DOUBLE, DT_INT32, DT_UINT8, DT_INT16, ..., DT_QUINT8, DT_QINT32, DT_BFLOAT16, DT_COMPLEX128, DT_HALF]
 [Op:Equal]
```

Stackverflow question: [stackoverflow](https://stackoverflow.com/questions/61751605/compare-tensor-of-unsigned-int-to-python-int/61752565#61752565)

**Comment by yongtang** [comment]
@zabir-nabil The issue has been fixed in PR #38288 and is available in tf-nightly.

**Comment by zabir-nabil** [comment]
Okay, great. So, the tf-nightly works.
